src/helm/benchmark/annotation/model_as_judge_annotator.py

```diff
@@ -0,0 +1,52 @@
+import json
+import os
+from typing import Dict
+
+from helm.common.general import ensure_directory_exists
+from helm.benchmark.annotation.annotator import Annotator
+from helm.clients.auto_client import AutoClient
+from helm.common.request import Request
+
+
+class ModelAsJudgeAnnotator(Annotator):
+    """Model as judge autograder."""
+
+    def __init__(self, auto_client: AutoClient, file_storage_path: str):
+        self._auto_client = auto_client
+        cache_dir = os.path.join(file_storage_path, "data")
+        ensure_directory_exists(cache_dir)
+        super().__init__()
+
+    # expects a prompt that includes a user request that ensures the model needs to return a score and reasoning
+    def score_with_reasoning(
+        self,
+        annotator_prompt: str,
+        annotator_model: str,
+        annotator_model_deployment: str,
+    ) -> Dict:
+        annotator_request = Request(
+            model=annotator_model,
+            model_deployment=annotator_model_deployment,
+            prompt=annotator_prompt,
+            temperature=0.0,
+            max_tokens=256,
+        )
+        annotator_response = self._auto_client.make_request(annotator_request)
+        if not annotator_response.success:
+            raise Exception(f"Annotation request failed: {annotator_response.error}")
+        assert len(annotator_response.completions) == 1
+        annotator_response_text = annotator_response.completions[0].text
+        try:
+            parsed_response = json.loads(annotator_response_text)
```

Comment:
GPT likes to respond with markdown backticks; you might need to parse it out like so: https://github.com/stanford-crfm/helm/blob/5021be1b5965dcc71fa2b0f394fd338fd3be2edf/src/helm/benchmark/annotation/financebench_annotator.py#L70-L71